mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/StatefulPuzzleGameScreenTest.kt

```diff
@@ -0,0 +1,304 @@
+package ch.epfl.sdp.mobile.test.state
+
+import androidx.compose.ui.test.*
+import androidx.compose.ui.test.junit4.createComposeRule
+import ch.epfl.sdp.mobile.application.authentication.AuthenticatedUser
+import ch.epfl.sdp.mobile.application.authentication.AuthenticationFacade
+import ch.epfl.sdp.mobile.application.chess.ChessFacade
+import ch.epfl.sdp.mobile.application.chess.engine.Delta
+import ch.epfl.sdp.mobile.application.chess.engine.Position
+import ch.epfl.sdp.mobile.application.chess.engine.Rank
+import ch.epfl.sdp.mobile.application.social.SocialFacade
+import ch.epfl.sdp.mobile.application.speech.SpeechFacade
+import ch.epfl.sdp.mobile.state.ProvideFacades
+import ch.epfl.sdp.mobile.state.StatefulGameScreenActions
+import ch.epfl.sdp.mobile.state.StatefulPuzzleGameScreen
+import ch.epfl.sdp.mobile.test.infrastructure.assets.fake.FakeAssetManager
+import ch.epfl.sdp.mobile.test.infrastructure.persistence.auth.emptyAuth
+import ch.epfl.sdp.mobile.test.infrastructure.persistence.store.emptyStore
+import ch.epfl.sdp.mobile.test.infrastructure.speech.FailingSpeechRecognizerFactory
+import ch.epfl.sdp.mobile.test.ui.game.ChessBoardRobot
+import ch.epfl.sdp.mobile.test.ui.game.click
+import ch.epfl.sdp.mobile.test.ui.game.play
+import ch.epfl.sdp.mobile.ui.game.ChessBoardState
+import com.google.accompanist.permissions.ExperimentalPermissionsApi
+import kotlinx.coroutines.flow.filterIsInstance
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.test.runTest
+import org.junit.Rule
+import org.junit.Test
+
+class StatefulPuzzleGameScreenTest {
+  @get:Rule val rule = createComposeRule()
+
+  @OptIn(ExperimentalPermissionsApi::class)
+  @Test
+  fun given_aPuzzleOnGameScreen_when_itIsDisplayed_then_expectedInfoDisplayed() = runTest {
+    val auth = emptyAuth()
+    val store = emptyStore()
+    val assets =
+        FakeAssetManager(
+            csvString =
+                "PuzzleId,FEN,Moves,Rating,RatingDeviation,Popularity,NbPlays,Themes,GameUrl\n" +
+                    "008Nz,6k1/2p2ppp/pnp5/B7/2P3PP/1P1bPPR1/r6r/3R2K1 b - - 1 29,d3e2 d1d8,600,101,85,298,backRankMate mate mateIn1 middlegame oneMove,https://lichess.org/HNU4zavC/black#58\n",
+        )
+
+    val authFacade = AuthenticationFacade(auth, store)
+    val socialFacade = SocialFacade(auth, store)
+    val chessFacade = ChessFacade(auth, store, assets)
+    val speechFacade = SpeechFacade(FailingSpeechRecognizerFactory)
+
+    authFacade.signUpWithEmail("email@example.org", "user", "password")
+    authFacade.signInWithEmail("email@example.org", "password")
```

Comment:
I think this `signIn` is not needed

Reply:
You're right, this was an artifact that came from the previous PR. Fixed in https://github.com/epfl-SDP/android/pull/309/commits/ef61afcc32032b6809c1fcac6daeed4655e051ef